### PR TITLE
A bit simplified code and update to Swift 5.1

### PR DESCRIPTION
**Pull request by Kn1kt**
### Checklist

- [x] I've looked at the [contribution guidelines](https://github.com/raywenderlich/swift-algorithm-club/blob/master/.github/CONTRIBUTING.md).
- [x] This pull request is complete and ready for review.

### Description
Fixed inconsistencies in Boyer-Moore-Horspool string search algorithm
Rewritten hash functions in Rabin-Karp string search algorithm
Simplified code in Knuth-Morris-Pratt String Search algorithm

### CODE REVIEW

File: Boyer-Moore-Horspool/BoyerMooreHorspool.swift [hunk 1]
```diff
@@ -6,70 +6,78 @@
...
-        func backwards() -> Index? {
-            var q = p
-            var j = i
-            while q > pattern.startIndex {
-                j = index(before: j)
-                q = index(before: q)
-                if self[j] != pattern[q] { return nil }
-            }
-            return j
+  func index(of pattern: String, usingHorspoolImprovement: Bool = false) -> Index? {
```
Comment (pkamb) [inline Boyer-Moore-Horspool/BoyerMooreHorspool.swift:9]: Indentation was changed from 4 spaces to 2 spaces, hiding any changes in this commit diff.

Style guide does say to use 2 spaces...

 > https://github.com/raywenderlich/swift-style-guide#spacing

But I would suggest fixing the indenting in a separate commit and/or PR, then making the algorithm changes in the next.

Comment (Kn1kt) [reply Boyer-Moore-Horspool/BoyerMooreHorspool.swift]: Now it's okay? Or maybe create one more PR?

Comment (pkamb) [reply Boyer-Moore-Horspool/BoyerMooreHorspool.swift]: That does now make the "Files Changed" diff much easier to read.

Personally I would re-create and re-push this branch to remove those indentation commits entirely from the git history, but others may have different opinions there. Not sure what the repo's standards are.

Comment (kelvinlauKL) [reply Boyer-Moore-Horspool/BoyerMooreHorspool.swift]: Thanks for pointing this out @pkamb

**Comment by RobertoMachorro** [comment]
is this pull abandoned or will someone be doing an update on the conflict?

**Comment by iiicey** [comment]
邮件已收到。我会尽快回复！祝您一切顺利！兰雨田

**Pull request closed without merging**